state/state.go

```diff
@@ -368,12 +367,22 @@ func (st *State) start(controllerTag names.ControllerTag) (err error) {
 
 	st.controllerTag = controllerTag
 
-	if identity := st.mongoInfo.Tag; identity != nil {
-		// TODO(fwereade): it feels a bit wrong to take this from MongoInfo -- I
-		// think it's just coincidental that the mongodb user happens to map to
-		// the machine that's executing the code -- but there doesn't seem to be
-		// an accessible alternative.
-		st.leaseClientId = identity.String()
+	var connectionStatus struct {
+		AuthInfo struct {
+			AuthenticatedUsers []struct {
+				User string `bson:"user"`
+			} `bson:"authenticatedUsers"`
+		} `bson:"authInfo"`
+	}
+	if err := st.session.DB(jujuDB).Run(bson.D{{"connectionStatus", 1}}, &connectionStatus); err != nil {
```

Comment:
Can you add a comment here? I don't quite grok this at first read.

Reply:
yup - sorry, done